DIO_Driver/DIO_Driver/Mcal/DIO/DIO.c

```diff
@@ -0,0 +1,285 @@
+{
+   /* check if invalid pin is given. */
+   if(PIN_7 < pin)
+   {
+      return E_DIO_INVALID_PIN;
+   }
+
+   /* pointers to carry the required port registers address. */
+   volatile uint8_t * prt_port_data = NULL_PTR;
+   
+   /* sets the data pointer according to the required port. */
+   switch(port)
+   {
+      case PORTA:
+         prt_port_data = &PORTA_DATA;
+         break;
+      case PORTB:
+         prt_port_data = &PORTB_DATA;
+         break;
+      case PORTC:
+         prt_port_data = &PORTC_DATA;
+         break;
+      case PORTD:
+         prt_port_data = &PORTD_DATA;
+         break;
+      default:
+         return E_DIO_INVALID_PORT;
+   }
+   /* set or clear value in the pin. */
+   if(HIGH == data)
+   {
+      SET_BIT(*prt_port_data,pin);
+   }
+   else if(LOW == data)
+   {
+      CLEAR_BIT(*prt_port_data,pin);
+   }
+   /* if data is invalid. */
+   else
+   {
+      return E_DIO_INVALID_WRITE_DATA;
+   }
+   
+   /* return success message. */
+   return E_DIO_SUCCESS;
+}
+
+/**
+* @brief: This function returns value of the pin in the given port.
+*
+* @param [in]  port     -  Port number.
+* @param [in]  pin      -  Pin number.
+* @param [out] data     -  pointer to the variable to save the pin value in it.
+*
+* @return function error state.
+*/
+DIO_ERROR_state_t DIO_ReadPin(uint8_t port, uint8_t pin, uint8_t * data)
+{
+   /* pointers to carry the required port registers address. */
+   volatile uint8_t * prt_port_stat = NULL_PTR;
+   
+   /* check if address of data to put the read value in is not null pointer. */
+   if(NULL_PTR == data)
+   {
+      return E_DIO_NULL_PTR;
+   }
+   /* check if invalid pin is given. */
+   else if(PIN_7 < pin)
+   {
+      return E_DIO_INVALID_PIN;
+   }
+   else
+   {
+      /* do nothing */
+   }
+   
+   /* sets the stat pointer according to the required port. */
+   switch(port)
+   {
+      case PORTA:
+         prt_port_stat = &PORTA_STAT;
+         break;
+      case PORTB:
+         prt_port_stat = &PORTB_STAT;
+         break;
+      case PORTC:
+         prt_port_stat = &PORTC_STAT;
+         break;
+      case PORTD:
+         prt_port_stat = &PORTD_STAT;
+         break;
+      default:
+         return E_DIO_INVALID_PORT;
+   }
+   
+   /* read the pin value and put it in the data. */
+   if(BIT_IS_SET(*prt_port_stat,pin))
+   {
+      *data = HIGH;
+   }
+   else
+   {
+      *data = LOW;
+   }
+   
+   /* return success message. */
+   return E_DIO_SUCCESS;
+}
+
+/**
+* @brief: This function toggles the value of the pin in the given port.
+*
+* @param [in]  port     -  Port number.
+* @param [in]  pin      -  Pin number.
+*
+* @return function error state.
+*/
+DIO_ERROR_state_t DIO_TogglePin(uint8_t port, uint8_t pin)
+{
+   /* check if invalid pin is given. */
+   if(PIN_7 < pin)
+   {
+      return E_DIO_INVALID_PIN;
+   }
+
+   /* pointers to carry the required port registers address. */
+   volatile uint8_t * prt_port_data = NULL_PTR;
```

Comment:
-local variable defination:
You are not following Sprints_CodeGuideLine 
![1](https://user-images.githubusercontent.com/44152298/127495302-babfec24-84d0-473b-9594-48095972fd28.PNG)